x-pack/plugins/cloud_security_posture/public/components/cloud_security_data_table/fields_selector/fields_selector_modal.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { type DataView } from '@kbn/data-views-plugin/common';
+import useSessionStorage from 'react-use/lib/useSessionStorage';
+import { SESSION_STORAGE_FIELDS_MODAL_SHOW_SELECTED } from '../../../common/constants';
+import { FieldsSelectorTable } from './fields_selector_table';
+import {
+  CSP_FIELDS_SELECTOR_CLOSE_BUTTON,
+  CSP_FIELDS_SELECTOR_MODAL,
+  CSP_FIELDS_SELECTOR_RESET_BUTTON,
+} from '../../test_subjects';
+
+interface FieldsSelectorModalProps {
+  dataView: DataView;
+  columns: string[];
+  onAddColumn: (column: string) => void;
+  onRemoveColumn: (column: string) => void;
+  closeModal: () => void;
+  onResetColumns: () => void;
+}
+
+const title = i18n.translate('xpack.csp.dataTable.fieldsModalTitle', {
+  defaultMessage: 'Fields',
+});
+
+export const FieldsSelectorModal = ({
+  closeModal,
+  dataView,
+  columns,
+  onAddColumn,
+  onRemoveColumn,
+  onResetColumns,
+}: FieldsSelectorModalProps) => {
+  const [isFilterSelectedEnabled, setIsFilterSelectedEnabled] = useSessionStorage(
```

Comment:
using session storage as a simple UX enhancement over useState (being able to see your last selected option), and a more permanent storage as local storage is not necessary